docs/IAS-XSUAA-token-xchange.md

```diff
@@ -0,0 +1,66 @@
+# IAS to XSUAA Token exchange
+Token exchange supports principal propagation from an IAS-based app/service to an XSUAA-based service, logically staying in the same zone.
+This is a hybrid scenario of cross consumption between IAS and XSUAA.
+
+### Scenario
+The user is logged in to a multitenant application that has adopted IAS and zones.
+The application (or an internally used SCP service instance) calls a service in the context of the current user and zone, by sending the IAS ID token for the user. That means the IAS ID token consists of `user_uuid` and `zone_uuid` claim.
+The service instance uses XSUAA internally and thus needs to exchange the ID token from IAS to an access token from XSUAA. 
+In the XSUAA multitenancy concept, the zone is mapped to the single XSUAA tenant that belongs to the zone (=> XSUAA tenant ID = zone ID).
+
+### Under the hood
+![IAS -> XSUAA token xchange flow diagram](token-xchange.png)
+
+Token of incoming request is checked whether it is a Xsuaa token, if it is Xsuaa token then this token proceeds forward to the token validation step. In case it is valid token, successful response is sent back, otherwise an unauthorised response is sent. 
+In case token is not Xsuaa token, application checks if token exchange is enabled, if it is, then Xsuaa access token is requested from Xsuaa instance using the POST request with Header `X-zid`:`zone_uuid`, `grant_type` = `jwt-bearer` and `assertion` = IAS token from the request. Upon successful token exchange between IAS and XSUAA, token gets validated and depending on validation result authorised or unauthorised response is sent back.
+In situation when incoming request doesn't contain Xsuaa token and token exchange is disabled, unauthorised response is sent back. 
+### Setup
+#### Setup trust between IAS and Xsuaa 
+Detailed information can be found [here](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/161f8f0cfac64c4fa2d973bc5f08a894.html)
+1. create an XSUAA service instance with plan apiaccess in your subaccount
+2. fetch access token from this service instance
+3. execute the trust setup:
+  ```shell script
+    curl --location --request POST 'https://api.authentication.sap.hana.ondemand.com/sap/rest/identity-providers' \
+    --header 'Authorization: bearer <FETCHED XSUAA ACCESS TOKEN>' \
```

Comment:
is curl still required, when you did as explained in help.sap.com? Maybe you can ask Patrick?

Reply:
confirmed by Patrick both do the same thing. Thanks for pointing out 💯